### PR TITLE
[make:registration] cleanup maker and template

### PR DESCRIPTION
**Pull request by jrushlow**
- Implements a new use statement sorter for templates
- use class properties instead of abusing command arguments

### CODE REVIEW

File: src/Generator.php [hunk 1]
```diff
@@ -239,4 +239,9 @@ public function generateTemplate(string $targetPath, string $templateName, array
             $variables
         );
     }
+
+    public static function getParentClassName(): string
```
Comment (weaverryan) [inline src/Generator.php:243]:
```suggestion
    public static function getControllerBaseClass(): string
```

And really, this should return the FQCN. If we need to tweak `generateController()` to only grab the "last" piece, then let's do that.

File: src/Util/TemplateComponentGenerator.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Util;
+
+/**
+ * @author Jesse Rushlow <jr@rushlow.dev>
+ *
+ * @internal
```
Comment (weaverryan) [inline src/Util/TemplateComponentGenerator.php:17]: After we've used this a bunch and finalized it, we should consider removing the `@internal`.

Comment (jrushlow) [reply src/Util/TemplateComponentGenerator.php:17]: agreed - I'm thinking that once we roll out the other pieces to the "new template generator", make all of those components available for use in userland code.

File: src/Generator.php (hunk 1 above)
Comment (jrushlow) [reply src/Generator.php]: this I've come up with a more elegant solution than depending on a string.

Comment (jrushlow) [reply src/Generator.php]: done

**Comment by weaverryan** [comment]
Thanks Jesse!

**Pull request merged**